### PR TITLE
join reorder: add missing part for dp algorithm

### PR DESCRIPTION
**Pull request by winoros**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add doc for dp algorithm of join reorder

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: join-reorder.md [hunk 1]
```diff
@@ -7,9 +7,9 @@ aliases: ['/docs-cn/dev/join-reorder/','/docs-cn/dev/reference/performance/join-
 
 在实际的业务场景中，多个表的 Join 语句是很常见的，而 Join 的执行效率和各个表参与 Join 的顺序有关系。如 `select * from t1, t2, t3 where t1.a=t2.a and t3.a=t2.a`，这个 SQL 中可能的执行顺序有“t1 和 t2 先做 Join，然后再和 t3 做 Join”以及“t2 和 t3 先做 Join，然后再和 t1 做 Join”两种情况。根据 `t1` 和 `t3` 的数据量及数据分布，这两种执行顺序会有不同的性能表现。
 
-因此优化器需要实现一种决定 Join 顺序的算法。目前 TiDB 中使用的算法是 Join Reorder 算法，又称贪心算法。
+因此优化器需要实现一种决定 Join 顺序的算法。目前 TiDB 中存在两种 Join Reorder 算法，贪心算法和动态规划算法。
```
Comment (qiancai) [inline join-reorder.md:11]: 这里是否可以加一句，简短解释下什么是贪心算法和 动态规划算法，然后下面再提供实例

File: join-reorder.md [hunk 2]
```diff
@@ -7,9 +7,9 @@ aliases: ['/docs-cn/dev/join-reorder/','/docs-cn/dev/reference/performance/join-
...
```
Comment (qiancai) [inline join-reorder.md:10]:
```suggestion
因此优化器需要实现一种决定 Join 顺序的算法。目前 TiDB 中存在两种 Join Reorder 算法，贪心算法和动态规划算法。

- Join Reorder 贪心算法： 在所有参与 Join 的节点中，选择行数最小的表与其他各表分别做一次 Join 的结果估算，然后选择其中结果最小的一对进行 Join，再继续这个过程进入下一轮的选择和 Join，直到所有的节点都完成 Join。
- Join Reorder 动态规划算法：在所有参与 Join 的节点中，枚举所有可能的 Join 顺序，然后选择最优的 Join 顺序。
```

File: join-reorder.md [hunk 3]
```diff
@@ -23,13 +23,28 @@ aliases: ['/docs-cn/dev/join-reorder/','/docs-cn/dev/reference/performance/join-
...
+
+![join-reorder-4](/media/join-reorder-4.png)
+
+当该选择比贪心算法更优时，动态规划算法便可以选择到更优的 Join 顺序。
+
+相应的，因为会枚举所有的可能性，动态规划算法会消耗更多的时间，也会更容易受统计信息影响。
+
+## Join Reorder 算法的控制
+
+目前 Join Reorder 算法由变量 [tidb_opt_join_reorder_threshold](/system-variables.md#tidb_opt_join_reorder_threshold) 控制，当参与 Join Reorder 的节点个数大于该阈值时选择贪心算法，反之选择动态规划算法。
```
Comment (Oreoxmt) [inline join-reorder.md:40]:
```suggestion
目前 Join Reorder 算法由变量 [`tidb_opt_join_reorder_threshold`](/system-variables.md#tidb_opt_join_reorder_threshold) 控制，当参与 Join Reorder 的节点个数大于该阈值时选择贪心算法，反之选择动态规划算法。
```

File: join-reorder.md [hunk 4]
```diff
@@ -23,13 +26,28 @@ aliases: ['/docs-cn/dev/join-reorder/','/docs-cn/dev/reference/performance/join-
 
 ![join-reorder-3](/media/join-reorder-3.png)
 
-以上就是当前 TiDB 中使用的 Join reorder 算法。
+以上就是当前 TiDB 中使用的 Join reorder 贪心算法。
```
Comment (Oreoxmt) [inline join-reorder.md:29]:
```suggestion
以上就是当前 TiDB 中使用的 Join Reorder 贪心算法。
```

File: join-reorder.md [hunk 5]
```diff
@@ -23,13 +26,28 @@ aliases: ['/docs-cn/dev/join-reorder/','/docs-cn/dev/reference/performance/join-
...
+
+## Join Reorder 动态规划算法实例
+
+仍然以上述例子为例。动态规划算法会枚举所有的可能性，因此相对贪心算法必须从 `t1` 表开始枚举，动态规划算法可以枚举如下的 Join 顺序。
+
+![join-reorder-4](/media/join-reorder-4.png)
+
+当该选择比贪心算法更优时，动态规划算法便可以选择到更优的 Join 顺序。
+
+相应的，因为会枚举所有的可能性，动态规划算法会消耗更多的时间，也会更容易受统计信息影响。
```
Comment (Oreoxmt) [inline join-reorder.md:39]:
```suggestion
相应地，因为会枚举所有的可能性，动态规划算法会消耗更多的时间，也会更容易受统计信息影响。
```

File: join-reorder.md [hunk 6]
```diff
@@ -23,13 +26,28 @@ aliases: ['/docs-cn/dev/join-reorder/','/docs-cn/dev/reference/performance/join-
...
+
+## Join Reorder 算法的控制
+
+目前 Join Reorder 算法由变量 [`tidb_opt_join_reorder_threshold`](/system-variables.md#tidb_opt_join_reorder_threshold) 控制，当参与 Join Reorder 的节点个数大于该阈值时选择贪心算法，反之选择动态规划算法。
 
 ## Join reorder 算法限制
```
Comment (Oreoxmt) [inline join-reorder.md:45]:
```suggestion
## Join Reorder 算法限制
```

File: join-reorder.md [hunk 7]
```diff
@@ -23,13 +26,28 @@ aliases: ['/docs-cn/dev/join-reorder/','/docs-cn/dev/reference/performance/join-
...
 
 当前的 Join Reorder 算法存在如下限制：
 
 - 受结果集的计算算法所限并不会保证一定会选到合适的 Join order
 - 目前默认关闭 Outer Join 的 Join Reorder。如需开启此功能，需要将系统变量 [`tidb_enable_outer_join_reorder`](/system-variables.md#tidb_enable_outer_join_reorder-从-v610-版本开始引入) 设为 `ON`。
+- 动态规划算法目前还无法进行 Outer Join 的 Join Reorder。
```
Comment (Oreoxmt) [inline join-reorder.md:51]:
```suggestion
- 目前动态规划算法无法进行 Outer Join 的 Join Reorder。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

File: join-reorder.md [hunk 8]
```diff
@@ -23,13 +26,28 @@ aliases: ['/docs-cn/dev/join-reorder/','/docs-cn/dev/reference/performance/join-
...
-## Join reorder 算法限制
+## Join Reorder 算法限制
 
 当前的 Join Reorder 算法存在如下限制：
 
 - 受结果集的计算算法所限并不会保证一定会选到合适的 Join order
 - 目前默认关闭 Outer Join 的 Join Reorder。如需开启此功能，需要将系统变量 [`tidb_enable_outer_join_reorder`](/system-variables.md#tidb_enable_outer_join_reorder-从-v610-版本开始引入) 设为 `ON`。
+- 目前动态规划算法无法进行 Outer Join 的 Join Reorder。
 
 目前 TiDB 中支持使用 `STRAIGHT_JOIN` 语法来强制指定一种 Join 顺序，参见[语法元素说明](/sql-statements/sql-statement-select.md#语法元素说明)。
```
Comment (Oreoxmt) [inline join-reorder.md:53]: media/join-reorder-1.png、media/join-reorder-3.png、media/join-reorder-4.png 建议将这几张图片的宽度调小，空白部分过多文档站渲染出来会有大量空白，对用户来说单屏的有效信息量较少

_预览效果 join-reorder-1.png_
image

_预览效果 join-reorder-3.png_

image

_预览效果 join-reorder-4.png_

image

File: join-reorder.md [hunk 9]
```diff
@@ -23,13 +26,28 @@ aliases: ['/docs-cn/dev/join-reorder/','/docs-cn/dev/reference/performance/join-
 
 ![join-reorder-3](/media/join-reorder-3.png)
 
-以上就是当前 TiDB 中使用的 Join reorder 算法。
+以上就是当前 TiDB 中使用的 Join Reorder 贪心算法。
+
```
Comment (Oreoxmt) [inline join-reorder.md:30]:
```suggestion
```

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

**Review by qiancai — APPROVED** [review]
LGTM

**Pull request merged**